app/src/main/java/io/appium/settings/Settings.java

```diff
@@ -70,12 +94,146 @@ public void onCreate(Bundle savedInstanceState) {
             LocationTracker.getInstance().start(this);
         }
 
+        handleRecording(getIntent());
+    }
+
+    private void handleRecording(Intent intent) {
+        if (intent == null) {
+            Log.e(TAG, "handleRecording: Unable to retrieve intent instance");
+            finishActivity();
+            return;
+        }
+
+        String recordingAction = intent.getAction();
+        if (recordingAction == null) {
+            Log.e(TAG, "handleRecording: Unable to retrieve intent.action instance");
+            finishActivity();
+            return;
+        }
+
+        if (!recordingAction.startsWith(ACTION_RECORDING_BASE)) {
+            Log.i(TAG, "handleRecording: Received different intent with action: "
+                    + recordingAction);
+            finishActivity();
+            return;
+        }
+
+        if (RecorderUtil.isLowerThanQ()) {
+            Log.e(TAG, "handleRecording: Current Android OS Version is lower than Q");
+            finishActivity();
+            return;
+        }
+
+        if (!RecorderUtil.areRecordingPermissionsGranted(getApplicationContext())) {
+            Log.e(TAG, "handleRecording: Required permissions are not granted");
+            finishActivity();
+            return;
+        }
+
+        if (recordingAction.equals(ACTION_RECORDING_START)) {
+            String recordingFilename = intent.getStringExtra(ACTION_RECORDING_FILENAME);
+            if (!RecorderUtil.isValidFileName(recordingFilename)) {
+                Log.e(TAG, "handleRecording: Invalid filename passed by user: "
+                        + recordingFilename);
+                finishActivity();
+                return;
+            }
+
+        /*
+        External Storage File Directory for app
+        (i.e /storage/emulated/0/Android/data/io.appium.settings/files) may not be created
+        so we need to call getExternalFilesDir() method twice
+        source:https://www.androidbugfix.com/2021/10/getexternalfilesdirnull-returns-null-in.html
+         */
+            File externalStorageFile = getExternalFilesDir(null);
+            if (externalStorageFile == null) {
+                externalStorageFile = getExternalFilesDir(null);
+            }
+            // if path is still null despite calling method twice, early exit
+            if (externalStorageFile == null) {
+                Log.e(TAG, "handleRecording: Unable to retrieve external storage file path");
+                finishActivity();
+                return;
+            }
+            recordingOutputPath = externalStorageFile.getAbsolutePath()
```

Comment:
you could also use https://stackoverflow.com/questions/711993/does-java-have-a-path-joining-method or https://developer.android.com/reference/java/nio/file/Paths

Reply:
done https://github.com/sirmordred/io.appium.settings/commit/991c57ebd035ea691191c3cf60a01d414cd867ea